### PR TITLE
feature/mutate: Mutation queries support pull request

### PR DESCRIPTION
**Pull request by philjung94**
Allows use of mutation schema and resolvers through the same interface as the previous GQL module. 
The exported class from entry file now has a class method called 'mutate' which can be used for accessing mutations. 

Some minor refactor conducted as well. 

Usage example:
`
const l = new LivingTrello(MY_KEY, MY_TOKEN)
l.mutate(`
  {
    getList(byId:MY_LIST_ID) {
      card1: addCard(withName: "Hello world") {
        id
        name
      }
      card2: addCard(withName: "Hello world2") {
        id
        name
      }
    }
  }
`).then(console.log).catch(console.log)

`

### CODE REVIEW

File: living-trello/src/index.js [hunk 1]
```diff
@@ -9,30 +9,44 @@
...
-      return graphql(schema, q, resolvers, {Trello: t}).then(({data}) => data)
-    } else {
-      throw new Error('Must set key and token to authenticate Trello via constructor(key, token).')
+    function lt(key, token) {
+      t = new Trello(key, token) 
     }
-  }
 
-  return lt
+    ['query', 'mutate'].forEach(cv => {
```
Comment (kevjy) [inline living-trello/src/index.js]: Can't we just do something like
```javascript
Object.values(docs).forEach(doc => {
```

File: living-trello/src/index.js [hunk 2]
```diff
@@ -9,30 +9,44 @@
...
-import {graphql} from 'graphql'
-import schema from './schema'
-import resolvers from './resolvers'
+import { graphql } from 'graphql'
+import query from './query'
+import mutate from './mutate'
 import Trello from './trello'
 
-const LivingTrello = (function(Trello, graphql, schema, resolvers) {
+const docs = {
```
Comment (kevjy) [inline living-trello/src/index.js]: Can you explain why these are called "docs"?

File: living-trello/src/query/schema.js [hunk 3]
```diff
@@ -12,7 +12,6 @@ export default buildSchema(`
   type List {
     id: ID!
     name: String!
-    desc: String!
```
Comment (kevjy) [inline living-trello/src/query/schema.js]: Why is this removed?

File: living-trello/src/trello.js [hunk 4]
```diff
@@ -5,32 +5,42 @@ export default class Trello {
...
+          url, 
+          method, 
+          body
+        }
+      )
+    ).catch(console.error)
   }
   /**
    * Parses the query string to ensure that excess chars are removed before passing it back to Trello.call
    * @param {String} query - the endpoint string without hostname, e.g. "/boards"
```
Comment (kevjy) [inline living-trello/src/trello.js]: doc needs updating

File: living-trello/src/trello.js [hunk 5]
```diff
@@ -5,32 +5,42 @@ export default class Trello {
...
+  urlFrom(path) {
+    let qs = path
+    if (path.charAt(0) === '/') {
+      qs = path.slice(1)
     }
     if (['&','?'].some(char => char === qs.charAt(qs.length-1))) {
       qs = qs.slice(0, qs.length-1)
     }
-    return qs
+    return `https://api.trello.com/1/${qs}${path.includes('?')? '&':'?'}key=${this.key}&token=${this.token}&card_limit=1000`
```
Comment (kevjy) [inline living-trello/src/trello.js]: path includes question mark logic doesn't look right

File: living-trello/src/trello.js [hunk 6]
```diff
@@ -5,32 +5,42 @@ export default class Trello {
...
   }
   /**
    * Promise executor with resolve & reject callbacks
    * Returns the callback for handling the https GET request
    * @param {Function} resolve, reject - from Promise
```
Comment (kevjy) [inline living-trello/src/trello.js]: Update docs

File: living-trello/src/trello.js [hunk 7]
```diff
@@ -45,6 +55,7 @@ export default class Trello {
       let rawData = ''
       res.on('data', (chunk) => { rawData += chunk })
       res.on('end', () => {
+        // console.log(rawData)
```
Comment (kevjy) [inline living-trello/src/trello.js]: Remove this?

File: living-trello/src/index.js (hunk 2 above)
Comment (philjung94) [reply living-trello/src/index.js]: Well we could call it `op` for operation but each op will contain two nested documents `schema` and `resolvers`

File: living-trello/src/index.js (hunk 1 above)
Comment (philjung94) [reply living-trello/src/index.js]: Yeap good catch

File: living-trello/src/index.js [hunk 8]
```diff
@@ -9,30 +9,39 @@
...
+        }
+      }
+    })
+
+    return lt
+
+  }
 
-})(Trello, graphql, schema, resolvers)
+)( Trello, graphql, query, mutate )
```
Comment (philjung94) [inline living-trello/src/index.js]: Let's keep the closure pattern but pass in our query and mutate operations

File: living-trello/src/query/schema.js (hunk 3 above)
Comment (philjung94) [reply living-trello/src/query/schema.js]: Not needed?

**Comment by philjung94** [comment]
@kevjy Please review.

File: living-trello/package.json [hunk 9]
```diff
@@ -16,6 +16,7 @@
   "dependencies": {
     "bluebird": "^3.5.5",
     "graphql": "^14.3.1",
+    "graphql-tools": "^4.0.5",
```
Comment (kevjy) [inline living-trello/package.json]: unused

File: living-trello/src/index.js [hunk 10]
```diff
@@ -9,30 +9,39 @@
...
+const LivingTrello = (
 
-  var t
+  // Deny access to our Trello Node.js interface. It's intended for use via GraphQL interface. 
+  function closure( Trello, graphql, query, mutate ) {
 
-  function lt(key, token) {
-    t = new Trello(key, token) // Deny access to our Trello Node.js interface. It's intended for use via GraphQL interface. 
-  }
+    var t, ops = {query, mutate} // Each op is an Object with nested documents {schema: Schema, resolvers: Resolvers}
```
Comment (kevjy) [inline living-trello/src/index.js]: `ops` can be `const`

File: living-trello/src/index.js [hunk 11]
```diff
@@ -9,30 +9,39 @@
...
-      return graphql(schema, q, resolvers, {Trello: t}).then(({data}) => data)
-    } else {
-      throw new Error('Must set key and token to authenticate Trello via constructor(key, token).')
+    function lt(key, token) {
+      t = new Trello(key, token) 
     }
-  }
 
-  return lt
+    Object.values(ops).forEach(cv => {
```
Comment (kevjy) [inline living-trello/src/index.js]: Not 100% sure about exact syntax but I'm thinking we should be able to do something like
```javascript
Object.entries(ops).forEach((name, op) => {
  lt.prototype[name] = function(q) {
    ...
    return graphql(op['schema'], q, ...
  }
})
```

File: living-trello/src/query/schema.js (hunk 3 above)
Comment (kevjy) [reply living-trello/src/query/schema.js]: If lists don't have description, we should update `query/resolvers.js` to not try to destructure a list into id props which contains desc
